### PR TITLE
DEV2-3173 fix smtp remove providers

### PR DESCRIPTION
**Pull request by amirbilu**

### CODE REVIEW

File: examples/complete/install.tf [hunk 1]
```diff
@@ -23,10 +23,15 @@ module "gke_cluster_tabnine_install" {
   exclude_kubernetes_manifest = var.exclude_kubernetes_manifest
   tabnine_registry_username   = var.tabnine_registry_username
   tabnine_registry_password   = var.tabnine_registry_password
-  smtp_pass                   = var.smtp_pass
-  smtp_user                   = var.smtp_user
+  smtp_auth_pass              = var.smtp_auth_pass
+  smtp_auth_user              = var.smtp_auth_user
```
Comment (amirbilu) [inline examples/complete/install.tf:27]: Naming convention was changed

File: examples/complete/install.tf [hunk 2]
```diff
@@ -23,10 +23,15 @@ module "gke_cluster_tabnine_install" {
...
   smtp_host                   = var.smtp_host
   smtp_ip                     = var.smtp_ip
   email_from_field            = var.email_from_field
   create_managed_cert         = var.create_managed_cert
+
+  depends_on = [
+    module.gke_cluster_tabnine
+  ]
```
Comment (amirbilu) [inline examples/complete/install.tf:35]: This is crucial for `terraform destroy` to work properly

File: examples/complete/providers.tf [hunk 3]
```diff
@@ -1,4 +1,20 @@
+data "google_client_config" "default" {}
```
Comment (amirbilu) [inline examples/complete/providers.tf:1]: Add providers to root module as suggested by terraform team. 

Providers in inner modules are considered deprecated

File: modules/install/prometheus_values.yaml.tftpl [hunk 4]
```diff
@@ -1,4 +1,6 @@
 defaultRules:
+      source  = "mycorp/mycloud"
```
Comment (yanir-codota) [inline modules/install/prometheus_values.yaml.tftpl:2]: reminder to check this

**Review by yanir-codota — APPROVED** [review]
LGTM

**Pull request merged**